lte/gateway/c/core/oai/tasks/amf/amf_recv.cpp

```diff
@@ -51,7 +51,6 @@ status_code_e amf_handle_service_request(
   uint16_t pdu_session_status = 0;
   uint32_t tmsi_stored;
   paging_context_t* paging_ctx = nullptr;
-  guti_and_amf_id_t guti_and_amf_id;
```

Comment:
In **line 303** `guti_and_amf_id_t guti_and_amf_id;` is marked by the compiler (and my IDE) as unused - but removing this variable makes `//lte/gateway/c/core/oai/test/amf:amf_procedures_test` fail with the following logging. If someone has an idea why, please explain :)
```
exec ${PAGER:-/usr/bin/less} "$0" || exit 1
Executing tests from //lte/gateway/c/core/oai/test/amf:amf_procedures_test
-----------------------------------------------------------------------------
Running main() from gmock_main.cc
[==========] Running 35 tests from 2 test suites.
[----------] Global test environment set-up.
[----------] 34 tests from AMFAppProcedureTest
[ RUN      ] AMFAppProcedureTest.TestRegistrationAuthSecurityModeReject
[       OK ] AMFAppProcedureTest.TestRegistrationAuthSecurityModeReject (1 ms)
[ RUN      ] AMFAppProcedureTest.TestRegistrationAuthSecurityCapabilityMismatch
[       OK ] AMFAppProcedureTest.TestRegistrationAuthSecurityCapabilityMismatch (0 ms)
[ RUN      ] AMFAppProcedureTest.TestRegistrationPlmnMismatch
[       OK ] AMFAppProcedureTest.TestRegistrationPlmnMismatch (0 ms)
[ RUN      ] AMFAppProcedureTest.TestRegistrationProcNoTMSI
[       OK ] AMFAppProcedureTest.TestRegistrationProcNoTMSI (1 ms)
[ RUN      ] AMFAppProcedureTest.TestDeRegistration
[       OK ] AMFAppProcedureTest.TestDeRegistration (2 ms)
[ RUN      ] AMFAppProcedureTest.TestRegistrationProcGutiBased
[       OK ] AMFAppProcedureTest.TestRegistrationProcGutiBased (0 ms)
[ RUN      ] AMFAppProcedureTest.TestRegistrationProcGutiBasedEncryption
[       OK ] AMFAppProcedureTest.TestRegistrationProcGutiBasedEncryption (1 ms)
[ RUN      ] AMFAppProcedureTest.TestMobileUpdatingRegistrationProcGutiBased
[       OK ] AMFAppProcedureTest.TestMobileUpdatingRegistrationProcGutiBased (0 ms)
[ RUN      ] AMFAppProcedureTest.TestPDUSessionSetup
[       OK ] AMFAppProcedureTest.TestPDUSessionSetup (1 ms)
[ RUN      ] AMFAppProcedureTest.TestPDUSessionSetupWithoutContext
[       OK ] AMFAppProcedureTest.TestPDUSessionSetupWithoutContext (1 ms)
[ RUN      ] AMFAppProcedureTest.TestPDUSessionSetupNoDnnNoSlice
[       OK ] AMFAppProcedureTest.TestPDUSessionSetupNoDnnNoSlice (1 ms)
[ RUN      ] AMFAppProcedureTest.TestPDUSessionFailure_dnn_not_subscribed
[       OK ] AMFAppProcedureTest.TestPDUSessionFailure_dnn_not_subscribed (0 ms)
[ RUN      ] AMFAppProcedureTest.TestPDUSession_missing_dnn
[       OK ] AMFAppProcedureTest.TestPDUSession_missing_dnn (1 ms)
[ RUN      ] AMFAppProcedureTest.TestPDUSession_unknown_pdu_session_type
[       OK ] AMFAppProcedureTest.TestPDUSession_unknown_pdu_session_type (0 ms)
[ RUN      ] AMFAppProcedureTest.TestPDUSession_Invalid_PDUSession_Identity
[       OK ] AMFAppProcedureTest.TestPDUSession_Invalid_PDUSession_Identity (1 ms)
[ RUN      ] AMFAppProcedureTest.TestRegistrationProcSUCIExt
[       OK ] AMFAppProcedureTest.TestRegistrationProcSUCIExt (0 ms)
[ RUN      ] AMFAppProcedureTest.TestAuthFailureFromSubscribeDb
[       OK ] AMFAppProcedureTest.TestAuthFailureFromSubscribeDb (1 ms)
[ RUN      ] AMFAppProcedureTest.TestPDUv6SessionSetup
[       OK ] AMFAppProcedureTest.TestPDUv6SessionSetup (0 ms)
[ RUN      ] AMFAppProcedureTest.TestPDUv4v6SessionSetup
[       OK ] AMFAppProcedureTest.TestPDUv4v6SessionSetup (1 ms)
[ RUN      ] AMFAppProcedureTest.ServiceRequestMTWithPDU
lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:1937: Failure
Expected equality of these values:
  true
  ue_context->pending_service_response
    Which is: false
lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:1940: Failure
Expected: (ue_context->amf_ue_ngap_id) != (ue_id), actual: 1 vs 1
lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:1977: Failure
Value of: expected_Ids == AMFClientServicer::getInstance().msgtype_stack
  Actual: false
Expected: true
[  FAILED  ] AMFAppProcedureTest.ServiceRequestMTWithPDU (1 ms)
[ RUN      ] AMFAppProcedureTest.ServiceRequestMTWithoutUplinkDataStatus
lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:2048: Failure
Value of: get_ue_id_from_imsi(amf_app_desc_p, imsi64, &updated_ue_id)
  Actual: false
Expected: true
lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:2050: Failure
Expected equality of these values:
  REGISTERED_CONNECTED
    Which is: 2
  ue_context->mm_state
    Which is: 1
lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:2053: Failure
Expected: (ue_context->amf_ue_ngap_id) != (ue_id), actual: 1 vs 1
lte/gateway/c/core/oai/test/amf/amf_app_test_util.cpp:278: Failure
Expected: (nullptr) != (ue_context_p), actual: (nullptr) vs NULL
lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:2062: Failure
Value of: rc == RETURNok
  Actual: false
Expected: true
lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:2066: Failure
Value of: expected_Ids == AMFClientServicer::getInstance().msgtype_stack
  Actual: false
Expected: true
[  FAILED  ] AMFAppProcedureTest.ServiceRequestMTWithoutUplinkDataStatus (1 ms)
[ RUN      ] AMFAppProcedureTest.ServiceRequestSignalling
lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:2136: Failure
Value of: get_ue_id_from_imsi(amf_app_desc_p, imsi64, &updated_ue_id)
  Actual: false
Expected: true
lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:2138: Failure
Expected equality of these values:
  REGISTERED_CONNECTED
    Which is: 2
  ue_context->mm_state
    Which is: 1
lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:2141: Failure
Expected: (ue_context->amf_ue_ngap_id) != (ue_id), actual: 1 vs 1
lte/gateway/c/core/oai/test/amf/amf_app_test_util.cpp:278: Failure
Expected: (nullptr) != (ue_context_p), actual: (nullptr) vs NULL
lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:2150: Failure
Value of: rc == RETURNok
  Actual: false
Expected: true
lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:2154: Failure
Value of: expected_Ids == AMFClientServicer::getInstance().msgtype_stack
  Actual: false
Expected: true
[  FAILED  ] AMFAppProcedureTest.ServiceRequestSignalling (0 ms)
[ RUN      ] AMFAppProcedureTest.ImplicitDeregDuplicateSuciReg
[       OK ] AMFAppProcedureTest.ImplicitDeregDuplicateSuciReg (1 ms)
[ RUN      ] AMFAppProcedureTest.ServiceRequestWrongTMSI
[       OK ] AMFAppProcedureTest.ServiceRequestWrongTMSI (1 ms)
[ RUN      ] AMFAppProcedureTest.ServiceRequestSignalWithPDU
lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:2562: Failure
Value of: get_ue_id_from_imsi(amf_app_desc_p, imsi64, &updated_ue_id)
  Actual: false
Expected: true
lte/gateway/c/core/oai/test/amf/amf_app_test_util.cpp:278: Failure
Expected: (nullptr) != (ue_context_p), actual: (nullptr) vs NULL
lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:2570: Failure
Value of: rc == RETURNok
  Actual: false
Expected: true
lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:2576: Failure
Value of: rc == RETURNok
  Actual: false
Expected: true
lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:2582: Failure
Value of: rc == RETURNok
  Actual: false
Expected: true
lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:2592: Failure
Value of: rc == RETURNok
  Actual: false
Expected: true
lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:2594: Failure
Value of: expected_Ids == AMFClientServicer::getInstance().msgtype_stack
  Actual: false
Expected: true
[  FAILED  ] AMFAppProcedureTest.ServiceRequestSignalWithPDU (1 ms)
[ RUN      ] AMFAppProcedureTest.PeriodicRegistraionNoTmsi
[       OK ] AMFAppProcedureTest.PeriodicRegistraionNoTmsi (1 ms)
[ RUN      ] AMFAppProcedureTest.ReRegistraion
lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:2777: Failure
Value of: expected_Ids == AMFClientServicer::getInstance().msgtype_stack
  Actual: false
Expected: true
[  FAILED  ] AMFAppProcedureTest.ReRegistraion (1 ms)
[ RUN      ] AMFAppProcedureTest.SctpShutWithServiceRequest
lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:2893: Failure
Value of: get_ue_id_from_imsi(amf_app_desc_p, imsi64, &updated_ue_id)
  Actual: false
Expected: true
lte/gateway/c/core/oai/test/amf/amf_app_test_util.cpp:278: Failure
Expected: (nullptr) != (ue_context_p), actual: (nullptr) vs NULL
lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:2901: Failure
Value of: rc == RETURNok
  Actual: false
Expected: true
lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:2907: Failure
Value of: rc == RETURNok
  Actual: false
Expected: true
lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:2913: Failure
Value of: rc == RETURNok
  Actual: false
Expected: true
lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:2923: Failure
Value of: rc == RETURNok
  Actual: false
Expected: true
lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:2925: Failure
Value of: expected_Ids == AMFClientServicer::getInstance().msgtype_stack
  Actual: false
Expected: true
[  FAILED  ] AMFAppProcedureTest.SctpShutWithServiceRequest (1 ms)
[ RUN      ] AMFAppProcedureTest.TestAuthFailureFromSubscribeDbLock
[       OK ] AMFAppProcedureTest.TestAuthFailureFromSubscribeDbLock (0 ms)
[ RUN      ] AMFAppProcedureTest.TestPDUSession_LocationUpdateFail
[       OK ] AMFAppProcedureTest.TestPDUSession_LocationUpdateFail (0 ms)
[ RUN      ] AMFAppProcedureTest.TestPDUSessionResourceModify
[       OK ] AMFAppProcedureTest.TestPDUSessionResourceModify (1 ms)
[ RUN      ] AMFAppProcedureTest.RegistrationAfterFourRegAcceptMsgs
lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:3199: Failure
Value of: get_ue_id_from_imsi(amf_app_desc_p, imsi64, &updated_ue_id)
  Actual: false
Expected: true
lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:3204: Failure
Expected: (ue_context->amf_ue_ngap_id) != (ue_id), actual: 1 vs 1
lte/gateway/c/core/oai/test/amf/amf_app_test_util.cpp:278: Failure
Expected: (nullptr) != (ue_context_p), actual: (nullptr) vs NULL
lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:3213: Failure
Value of: rc == RETURNok
  Actual: false
Expected: true
lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:3217: Failure
Value of: expected_Ids == AMFClientServicer::getInstance().msgtype_stack
  Actual: false
Expected: true
[  FAILED  ] AMFAppProcedureTest.RegistrationAfterFourRegAcceptMsgs (0 ms)
[ RUN      ] AMFAppProcedureTest.TestPDUSessionResourceModifyDeletion
[       OK ] AMFAppProcedureTest.TestPDUSessionResourceModifyDeletion (1 ms)
[ RUN      ] AMFAppProcedureTest.GnbInitiatedNGReset
[       OK ] AMFAppProcedureTest.GnbInitiatedNGReset (0 ms)
[----------] 34 tests from AMFAppProcedureTest (22 ms total)

[----------] 1 test from test_t3592abort
[ RUN      ] test_t3592abort.test_pdu_session_release_notify_smf
[       OK ] test_t3592abort.test_pdu_session_release_notify_smf (0 ms)
[----------] 1 test from test_t3592abort (0 ms total)

[----------] Global test environment tear-down
[==========] 35 tests from 2 test suites ran. (22 ms total)
[  PASSED  ] 28 tests.
[  FAILED  ] 7 tests, listed below:
[  FAILED  ] AMFAppProcedureTest.ServiceRequestMTWithPDU
[  FAILED  ] AMFAppProcedureTest.ServiceRequestMTWithoutUplinkDataStatus
[  FAILED  ] AMFAppProcedureTest.ServiceRequestSignalling
[  FAILED  ] AMFAppProcedureTest.ServiceRequestSignalWithPDU
[  FAILED  ] AMFAppProcedureTest.ReRegistraion
[  FAILED  ] AMFAppProcedureTest.SctpShutWithServiceRequest
[  FAILED  ] AMFAppProcedureTest.RegistrationAfterFourRegAcceptMsgs

 7 FAILED TESTS
```

Reply:
Just in case someone wants to dig deeper: This struct was added in [this change](https://github.com/ganeshg87/magma/blob/99f75cb1a2c549661dc01a733fdacacf12bc86b3/lte/gateway/c/core/oai/tasks/amf/amf_recv.cpp) in [this PR](https://github.com/magma/magma/pull/7565/files). Most of the relevant code was removed in [this PR](https://github.com/magma/magma/pull/11995/files#diff-4854cdf951944684bcf9438991e4a0df98bfdcc85686e1d99c384791e7bb2c22), with the exception of four definitions.

The tests pass even if the line `  guti_and_amf_id_t guti_and_amf_id;` is moved to line 623, the very end of the function `amf_handle_registration_request`. Therefore it is certainly not used inside of this function. Just adding a sleep also does not have an effect, so it is not just the time to create the struct that is relevant.